pallets/erc20-xcm-bridge/src/erc20_matcher.rs

```diff
@@ -67,6 +67,21 @@ impl<Erc20MultilocationPrefix: Get<MultiLocation>> Erc20Matcher<Erc20Multilocati
 			_ => Err(()),
 		}
 	}
+	pub(crate) fn matches_gas_limit(multiasset: &MultiAsset) -> Option<u64> {
+		let multilocation = match (&multiasset.fun, &multiasset.id) {
+			(Fungible(_), Concrete(ref id)) => id,
+			_ => return None,
+		};
+
+		let prefix = Erc20MultilocationPrefix::get();
+		match multilocation.interior().at(prefix.interior().len() + 1) {
+			Some(Junction::GeneralIndex { 0: value }) => match (*value).try_into() {
```

Comment:
Please don't use GeneralIndex, it's better to use a GeneralKey with an hardcoded prefix like `b"gas limit:"` followed by a be encoded u64 for instance.